demos/common/utilities/jws_using_mbedtls.c

```diff
@@ -0,0 +1,593 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include "jws.h"
+
+#include "azure/az_core.h"
+#include "azure/az_iot.h"
+
+#include "azure_iot_result.h"
+#include "azure_iot_json_reader.h"
+#include "azure_iot_adu_client.h"
+
+#include "mbedtls/base64.h"
+#include "mbedtls/rsa.h"
+#include "mbedtls/pk.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/cipher.h"
+
+/* For logging */
+#include "demo_config.h"
+
+/**
+ * @brief Convenience macro to return if an operation failed.
+ */
+#define azureiotresultRETURN_IF_FAILED( exp )        \
+    do                                               \
+    {                                                \
+        AzureIoTResult_t const _xAzResult = ( exp ); \
+        if( _xAzResult != eAzureIoTSuccess )         \
+        {                                            \
+            return _xAzResult;                       \
+        }                                            \
+    } while( 0 )
+
+#define jwsRSA3072_SIZE            384
+#define jwsSHA256_SIZE             32
+#define jwsPKCS7_PAYLOAD_OFFSET    19
+
+#define jwsSHA256_JSON_VALUE       "sha256"
+#define jwsSJWK_JSON_VALUE         "sjwk"
```

Comment:
can these be const uint8_t* jwsSJWK_JSON_VALUE = "sjwk";

Reply:
they can but having it this way allows for a `sizeof()` call